pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -291,41 +293,24 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr model.Config, event mo
 		return
 	}
 
-	if fullPush {
-		// When doing a full push, for added and updated services trigger an eds update
-		// so that endpoint shards are updated.
-		var instances []*model.ServiceInstance
-		if len(addedSvcs) > 0 {
-			instances = append(instances, convertInstances(curr, addedSvcs)...)
-		}
-		if len(updatedSvcs) > 0 {
-			instances = append(instances, convertInstances(curr, updatedSvcs)...)
-		}
-		if len(unchangedSvcs) > 0 {
-			currentServiceEntry := curr.Spec.(*networking.ServiceEntry)
-			oldServiceEntry := old.Spec.(*networking.ServiceEntry)
-			// Non DNS service entries are sent via EDS. So we should compare and update if such endpoints change.
-			if currentServiceEntry.Resolution != networking.ServiceEntry_DNS {
```

Comment:
@ramaraochavali here no matter what type of resolution, we need to update its instances. Take ServiceEntry_DNS as an example，we will build a cluster of `Cluster_STRICT_DNS` type with localityLbEndpoints. If not add here, we will miss them.

Reply:
But for STRICT_DNS we set host name and build locality lb endpoints in cluster here https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/cluster_builder.go#L238 ? Do we capture them eds shards? I do not think so

Reply:
You can continue trace the calling of `InstancesByPort`, which is based on serviceentry store. And the store is refreshed based on what we are doing here

Reply:
I guess it is working because of this check https://github.com/istio/istio/blob/master/pilot/pkg/xds/eds.go#L248 and we ignore building endpoints for it